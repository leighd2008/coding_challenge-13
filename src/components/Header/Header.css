header {
  transition: height 0.4s ease-in-out, background-color 0.4s ease-in-out,
    box-shadow 0.4s ease-in-out;
  height: 70px;
}

header.is-open,
header.scroll {
  background-color: rgba(255, 255, 255, 0.9);
  box-shadow: 0 2px var(--three);
}

nav {
  transition: height 0.5s ease-in-out, background-color 0.2s ease-in-out,
    box-shadow 0.4s ease-in-out;
  max-height: 70px;
}

.is-open nav {
  max-height: 500px;
}

.is-open .navbar {
  background-color: rgba(255, 255, 255, 0.9);
  border: 2px solid var(--three);
  border-top: 0;
  border-radius: 0px 0px 8px 8px;
}

.scroll {
  background: rgba(255, 255, 255, 0.9);
}

.navbar-collapse {
  transition: all 0.2s ease;
}
.scroll .navbar-collapse.show .navbar-nav {
}

.nav-buttons {
  margin-left: auto;
  padding: 15px;
}

.navbar-toggler {
  transition: all 0.4s;
  margin-left: auto;
  padding: 0.25rem;
}

.navbar-toggler:focus {
  outline: 0;
}

.navbar-toggler:hover {
  background-color: var(--three);
}

.navbar-toggler:hover .toggle-button-line {
  background: #fff;
}

.toggle-button-line {
  width: 1.25em;
  height: 2px;
  margin: 5px;
  background: grey;
}

#header a {
  color: var(--one);
}

#header .active {
  color: #fff;
}

#header a:hover {
  color: #fff;
}

.navbar-collapse #header a {
  color: rgba(255, 255, 255, 0);
  transition: color 0.6s ease-in-out;
}

.navbar-collapse.show #header a {
  color: var(--one);
}

.logo-link {
  /* display: inline-flex; */
  width: 40%;
  max-width: 250px;
}

.logo-active {
  transition: height 0.3s ease-in-out, width 0.3s ease-in-out,
    max-width 0.3s ease-in-out;
  cursor: pointer;
  min-width: 150px;
  max-width: 250px;
  width: auto;
  height: 55px;
}

.scroll .logo-active {
  height: 50px;
}
/* .smLogo > a > img {
  height: 40px;
  width: auto;
  position: relative;
  transition: height 0.3s ease-in-out;
} */

@media only screen and (max-width: 991px) {
  .nav-buttons a:first-child {
    margin-bottom: 7px;
  }
}
@media only screen and (max-width: 768px) {
  .logo-active {
    width: auto;
    height: 45px;
  }
  .scroll .logo-active {
    height: 40px;
  }
}
@media only screen and (max-width: 480px) {
  header {
    height: 60px;
  }
  .logo-active {
    width: auto;
    height: 40px;
  }
  .scroll .logo-active {
    height: 37px;
  }
}
