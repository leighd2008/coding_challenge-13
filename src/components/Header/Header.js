import React from 'react';
import { Link } from 'react-scroll';
import './Header.css';
import logo from '../../images/logo1.png';

const Header = ({headerClass}) => (
  <section>
    <nav id="header" className={headerClass}>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNavAltMarkup"
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <div className="toggle-button-line"></div>
		    <div className="toggle-button-line"></div>
        <div className="toggle-button-line"></div>
      </button>
      <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <Link to="top" smooth duration={800} offset={-131}>
            <img src={logo} alt="logo" height="50" className="logo-active" />
          </Link>
        <div className="navbar-nav nav-buttons">
          <Link
            to="newsletter"
            className="button-primary"
            spy
            activeClass="active"
            smooth
            duration={800}
            offset={0}
          >
            Newsletter
          </Link>
          <Link
            to="data"
            className="button-primary"
            spy
            activeClass="active"
            smooth
            duration={800}
            offset={0}
          >
            Data
          </Link>
        </div>
      </div>
    </nav>
  </section>
);

export default Header;
