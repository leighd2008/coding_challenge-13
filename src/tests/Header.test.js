import { shallow } from 'enzyme';
import React from 'react';
import Header from '../components/Header/Header';

it('expect to render Header component', () => {
	expect(shallow(<Header />)).toMatchSnapshot();
})