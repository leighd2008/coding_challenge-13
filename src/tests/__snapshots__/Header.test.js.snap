// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="navbar navbar-expand-md navbar-light fixed-top"
        id="header"
      >
        <Link
          duration={800}
          offset={-131}
          smooth={true}
          to="top"
        >
          <img
            alt="logo"
            className="logo-active"
            height="50"
            src="logo1.png"
          />
        </Link>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <div
            className="toggle-button-line"
          />
          <div
            className="toggle-button-line"
          />
          <div
            className="toggle-button-line"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav nav-buttons"
          >
            <Link
              activeClass="active"
              className="button-primary"
              duration={800}
              offset={0}
              smooth={true}
              spy={true}
              to="newsletter"
            >
              Newsletter
            </Link>
            <Link
              activeClass="active"
              className="button-primary"
              duration={800}
              offset={0}
              smooth={true}
              spy={true}
              to="data"
            >
              Data
            </Link>
          </div>
        </div>
      </nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            duration={800}
            offset={-131}
            smooth={true}
            to="top"
          >
            <img
              alt="logo"
              className="logo-active"
              height="50"
              src="logo1.png"
            />
          </Link>,
          <button
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarNavAltMarkup"
            data-toggle="collapse"
            type="button"
          >
            <div
              className="toggle-button-line"
            />
            <div
              className="toggle-button-line"
            />
            <div
              className="toggle-button-line"
            />
          </button>,
          <div
            className="collapse navbar-collapse"
            id="navbarNavAltMarkup"
          >
            <div
              className="navbar-nav nav-buttons"
            >
              <Link
                activeClass="active"
                className="button-primary"
                duration={800}
                offset={0}
                smooth={true}
                spy={true}
                to="newsletter"
              >
                Newsletter
              </Link>
              <Link
                activeClass="active"
                className="button-primary"
                duration={800}
                offset={0}
                smooth={true}
                spy={true}
                to="data"
              >
                Data
              </Link>
            </div>
          </div>,
        ],
        "className": "navbar navbar-expand-md navbar-light fixed-top",
        "id": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="logo"
              className="logo-active"
              height="50"
              src="logo1.png"
            />,
            "duration": 800,
            "offset": -131,
            "smooth": true,
            "to": "top",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "logo-active",
              "height": "50",
              "src": "logo1.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-controls": "navbarNavAltMarkup",
            "aria-expanded": "false",
            "aria-label": "Toggle navigation",
            "children": Array [
              <div
                className="toggle-button-line"
              />,
              <div
                className="toggle-button-line"
              />,
              <div
                className="toggle-button-line"
              />,
            ],
            "className": "navbar-toggler",
            "data-target": "#navbarNavAltMarkup",
            "data-toggle": "collapse",
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "toggle-button-line",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "toggle-button-line",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "toggle-button-line",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="navbar-nav nav-buttons"
            >
              <Link
                activeClass="active"
                className="button-primary"
                duration={800}
                offset={0}
                smooth={true}
                spy={true}
                to="newsletter"
              >
                Newsletter
              </Link>
              <Link
                activeClass="active"
                className="button-primary"
                duration={800}
                offset={0}
                smooth={true}
                spy={true}
                to="data"
              >
                Data
              </Link>
            </div>,
            "className": "collapse navbar-collapse",
            "id": "navbarNavAltMarkup",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  activeClass="active"
                  className="button-primary"
                  duration={800}
                  offset={0}
                  smooth={true}
                  spy={true}
                  to="newsletter"
                >
                  Newsletter
                </Link>,
                <Link
                  activeClass="active"
                  className="button-primary"
                  duration={800}
                  offset={0}
                  smooth={true}
                  spy={true}
                  to="data"
                >
                  Data
                </Link>,
              ],
              "className": "navbar-nav nav-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeClass": "active",
                  "children": "Newsletter",
                  "className": "button-primary",
                  "duration": 800,
                  "offset": 0,
                  "smooth": true,
                  "spy": true,
                  "to": "newsletter",
                },
                "ref": null,
                "rendered": "Newsletter",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeClass": "active",
                  "children": "Data",
                  "className": "button-primary",
                  "duration": 800,
                  "offset": 0,
                  "smooth": true,
                  "spy": true,
                  "to": "data",
                },
                "ref": null,
                "rendered": "Data",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "nav",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar navbar-expand-md navbar-light fixed-top"
          id="header"
        >
          <Link
            duration={800}
            offset={-131}
            smooth={true}
            to="top"
          >
            <img
              alt="logo"
              className="logo-active"
              height="50"
              src="logo1.png"
            />
          </Link>
          <button
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarNavAltMarkup"
            data-toggle="collapse"
            type="button"
          >
            <div
              className="toggle-button-line"
            />
            <div
              className="toggle-button-line"
            />
            <div
              className="toggle-button-line"
            />
          </button>
          <div
            className="collapse navbar-collapse"
            id="navbarNavAltMarkup"
          >
            <div
              className="navbar-nav nav-buttons"
            >
              <Link
                activeClass="active"
                className="button-primary"
                duration={800}
                offset={0}
                smooth={true}
                spy={true}
                to="newsletter"
              >
                Newsletter
              </Link>
              <Link
                activeClass="active"
                className="button-primary"
                duration={800}
                offset={0}
                smooth={true}
                spy={true}
                to="data"
              >
                Data
              </Link>
            </div>
          </div>
        </nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              duration={800}
              offset={-131}
              smooth={true}
              to="top"
            >
              <img
                alt="logo"
                className="logo-active"
                height="50"
                src="logo1.png"
              />
            </Link>,
            <button
              aria-controls="navbarNavAltMarkup"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler"
              data-target="#navbarNavAltMarkup"
              data-toggle="collapse"
              type="button"
            >
              <div
                className="toggle-button-line"
              />
              <div
                className="toggle-button-line"
              />
              <div
                className="toggle-button-line"
              />
            </button>,
            <div
              className="collapse navbar-collapse"
              id="navbarNavAltMarkup"
            >
              <div
                className="navbar-nav nav-buttons"
              >
                <Link
                  activeClass="active"
                  className="button-primary"
                  duration={800}
                  offset={0}
                  smooth={true}
                  spy={true}
                  to="newsletter"
                >
                  Newsletter
                </Link>
                <Link
                  activeClass="active"
                  className="button-primary"
                  duration={800}
                  offset={0}
                  smooth={true}
                  spy={true}
                  to="data"
                >
                  Data
                </Link>
              </div>
            </div>,
          ],
          "className": "navbar navbar-expand-md navbar-light fixed-top",
          "id": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="logo"
                className="logo-active"
                height="50"
                src="logo1.png"
              />,
              "duration": 800,
              "offset": -131,
              "smooth": true,
              "to": "top",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "logo-active",
                "height": "50",
                "src": "logo1.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-controls": "navbarNavAltMarkup",
              "aria-expanded": "false",
              "aria-label": "Toggle navigation",
              "children": Array [
                <div
                  className="toggle-button-line"
                />,
                <div
                  className="toggle-button-line"
                />,
                <div
                  className="toggle-button-line"
                />,
              ],
              "className": "navbar-toggler",
              "data-target": "#navbarNavAltMarkup",
              "data-toggle": "collapse",
              "type": "button",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "toggle-button-line",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "toggle-button-line",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "toggle-button-line",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="navbar-nav nav-buttons"
              >
                <Link
                  activeClass="active"
                  className="button-primary"
                  duration={800}
                  offset={0}
                  smooth={true}
                  spy={true}
                  to="newsletter"
                >
                  Newsletter
                </Link>
                <Link
                  activeClass="active"
                  className="button-primary"
                  duration={800}
                  offset={0}
                  smooth={true}
                  spy={true}
                  to="data"
                >
                  Data
                </Link>
              </div>,
              "className": "collapse navbar-collapse",
              "id": "navbarNavAltMarkup",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    activeClass="active"
                    className="button-primary"
                    duration={800}
                    offset={0}
                    smooth={true}
                    spy={true}
                    to="newsletter"
                  >
                    Newsletter
                  </Link>,
                  <Link
                    activeClass="active"
                    className="button-primary"
                    duration={800}
                    offset={0}
                    smooth={true}
                    spy={true}
                    to="data"
                  >
                    Data
                  </Link>,
                ],
                "className": "navbar-nav nav-buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "activeClass": "active",
                    "children": "Newsletter",
                    "className": "button-primary",
                    "duration": 800,
                    "offset": 0,
                    "smooth": true,
                    "spy": true,
                    "to": "newsletter",
                  },
                  "ref": null,
                  "rendered": "Newsletter",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "activeClass": "active",
                    "children": "Data",
                    "className": "button-primary",
                    "duration": 800,
                    "offset": 0,
                    "smooth": true,
                    "spy": true,
                    "to": "data",
                  },
                  "ref": null,
                  "rendered": "Data",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "nav",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
